File: package/src/tue_api_wrapper/alma_course_registration_models.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class AlmaCourseRegistrationOption:
    planelement_id: str
    label: str
    action_name: str


@dataclass(frozen=True)
class AlmaCourseRegistrationSupport:
    detail_url: str
    title: str | None
    number: str | None
    supported: bool
    action: str | None
    status: str | None = None
    messages: tuple[str, ...] = ()
    message: str | None = None


@dataclass(frozen=True)
class AlmaCourseRegistrationOptions:
    detail_url: str
    title: str | None
    number: str | None
    action: str
    options: tuple[AlmaCourseRegistrationOption, ...]
    messages: tuple[str, ...]


@dataclass(frozen=True)
class AlmaCourseRegistrationResult:
    detail_url: str
    final_url: str
    title: str | None
    number: str | None
    action: str
    selected_option: AlmaCourseRegistrationOption
    messages: tuple[str, ...]
    status: str | None

File: ios/Shared/Alma/AlmaCourseRegistrationModels.swift
import Foundation

struct AlmaCourseRegistrationOption: Equatable {
    var planelementID: String
    var label: String
    var actionName: String
}

struct AlmaCourseRegistrationSupport: Equatable {
    var detailURL: URL
    var supported: Bool
    var action: String?
    var status: String?
    var messages: [String]
    var message: String?
}

struct AlmaCourseRegistrationOptions: Equatable {
    var detailURL: URL
    var action: String
    var options: [AlmaCourseRegistrationOption]
    var messages: [String]
}

struct AlmaCourseRegistrationResult: Equatable {
    var detailURL: URL
    var finalURL: URL
    var action: String
    var selectedOption: AlmaCourseRegistrationOption
    var messages: [String]
    var status: String?
}

